wasm/src/record/plaintext_record.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the Aleo library.
+
+// The Aleo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Aleo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Aleo library. If not, see <https://www.gnu.org/licenses/>.
+
+use aleo_account::PlainTextRecord as PlainTextRecordNative;
+
+use std::{ops::Deref, str::FromStr};
+use wasm_bindgen::prelude::*;
+
+/// Aleo plaintext record
+#[wasm_bindgen]
+#[derive(Clone)]
+pub struct PlainTextRecord(PlainTextRecordNative);
+
+#[wasm_bindgen]
+impl PlainTextRecord {
+    /// Create a plaintext record object from a string
+    pub fn from_string(record: &str) -> Self {
+        Self::from_str(record).unwrap()
+    }
+
+    /// Get a string representation of the record
+    #[allow(clippy::inherent_to_string)]
+    pub fn to_string(&self) -> String {
+        self.0.to_string()
+    }
+
+    /// Get the amount of gates in the record
+    pub fn gates(&self) -> u64 {
+        ***self.0.gates()
+    }
+}
+
+impl From<PlainTextRecordNative> for PlainTextRecord {
+    fn from(record: PlainTextRecordNative) -> Self {
+        Self(record)
+    }
+}
+
+impl FromStr for PlainTextRecord {
+    type Err = anyhow::Error;
+
+    fn from_str(plaintext: &str) -> Result<Self, Self::Err> {
+        Ok(Self(PlainTextRecordNative::from_str(plaintext)?))
+    }
+}
+
+impl Deref for PlainTextRecord {
+    type Target = PlainTextRecordNative;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use wasm_bindgen_test::wasm_bindgen_test;
+
+    const RECORD: &str = "{ owner: aleo1d5hg2z3ma00382pngntdp68e74zv54jdxy249qhaujhks9c72yrs33ddah.private, gates: 99u64.public, _nonce: 0group.public }";
```

Comment:
convert to Rust raw string `r#`

Reply:
I believe what's causing the escape sequences to appear is Serde no-copy-deserializing from Rust objects which carry over escape sequences as artifacts. In SnarkVM all strings output have these artifacts even with `r#`. 

Thinking about the "to_string()" functionality, there's a few arguments for why one would have it.
1. To create human readable representations of the objects - this is of potential benefit in debugging, but also might be a security hole in a compromised system
2. To create output for command line inputs into leo programs - this has the *most* benefit IMO
3. To create JSON objects from - while seemingly useful upfront, I think this may actually promote poor usage patterns because it might start encouraging people to start peforming antipatterns like sending it over the wire in plaintext

The corresponding rust object doesn't have a debug derive I think exactly to protect the sensitive secrets in the record. I'm actually leaning towards not having a to_string() function and instead promoting an api which can create record usage in a way that hides more details.

Reply:
I believe what's causing the escape sequences to appear is Serde no-copy-deserializing from Rust objects which carry over escape sequences as artifacts. In SnarkVM all strings output have these artifacts even with r#, so I've changed the test a bit in next commit.

Thinking about the to_string() functionality, there's a few arguments for why one would have it:
* To create human readable representations of the objects - this is of potential benefit in debugging, but also might be a security hole in a compromised system
* To create output for command line inputs into leo programs - this has the most benefit IMO
* To create JSON objects from - while seemingly useful upfront, I think this may actually promote poor usage patterns because it might start encouraging people to start peforming antipatterns like sending it over the wire in plaintext

The corresponding rust object doesn't have a debug derive I think exactly to protect the sensitive secrets in the plaintext record. I'm actually leaning towards not having a to_string() function in the plaintext and instead promoting an api which can create record usage that has more clean security practices.